third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -22,12 +24,51 @@ var dataflowTerminalStatesMap = map[string]struct{}{
 	"JOB_STATE_DRAINED":   {},
 }
 
+func resourceDataflowJobLabelDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
+	if !strings.HasPrefix(k, "labels.") {
+		return false
+	}
+
+	// Example Diff: "labels.goog-dataflow-provided-template-version": "word_count" => ""
+	if strings.HasPrefix(k, resourceDataflowJobGoogleProvidedLabelPrefix) && new == "" {
+		// Suppress diff if field is a Google Dataflow-provided label key and has no explicitly set value in Config.
+		return true
+	}
+
+	// If k is comparing length of maps, compare old and new value.
+	if strings.HasPrefix(k, "labels.%") {
```

Comment:
What's an example where we would find a diff in this section but we wouldn't have in the logic above?

Reply:
An example diff that would cause this:
"labels.%": 2 => 0
"labels.google-dataflow-provided-template-name": "template" => ""
"labels.google-dataflow-provided-template-version": "2019_12_01_blahblah" => ""

The first if is just a guard so we aren't looking at non-label fields - I can remove it.
Second is for the two label fields.
Third is for the number comparison - we need to know whether new-old is empty and old-new is only suppressable labels.

Reply:
is just checking the label prefix enough?
like,
```
if strings.HasPrefix(k, "labels.%") {
  return true
}
```
and then getting rid of the other section? because then you'd return true in cases where it's the label prefix, and false in cases where it's a real diff, right?

Reply:
i believe so? There was a weird thing a long time ago where it didn't work with non-present vs empty maps, but it maybe got fixed by 0.12? I'm fine with removing it

Reply:
Ok! Yeah I don't really remember, but if you checked and this way works then yay less code.